src/gstreamer/gstkvssink.cpp

```diff
@@ -1664,15 +1661,8 @@ gst_kvs_sink_change_state(GstElement *element, GstStateChange transition) {
     // Downward transitions
     switch (transition) {
         case GST_STATE_CHANGE_PAUSED_TO_READY:
-            // Need this check in case an EOS was received in the buffer handler and
-            // stream was already stopped. Although stopSync() is an idempotent call,
-            // we want to avoid an extra call
-            if(!data->streamingStopped.load()) {
-                data->kinesis_video_stream->stopSync();
-                data->streamingStopped.store(true);
-            } else {
-                LOG_INFO("Streaming already stopped for " << kvssink->stream_name);
-            }
+            data->kinesis_video_stream->stopSync();
```

Comment:
suggest adding a log to denote trying to stop.

Reply:
This log is already there in the above switch block:

```
// (Downward Transition) gst_collect_pads_stop must be called prior to parent class PAUSED->READY transition.
case GST_STATE_CHANGE_PAUSED_TO_READY:
    LOG_INFO("Stopping kvssink for " << kvssink->stream_name);
```